src/extension.ts

```diff
@@ -171,41 +171,121 @@ class RunLineInEditor {
+            this.runningCommandCount += 1;
+            const t0 = Date.now();
+            if (this.runningCommandCount === 1)
+            {
+                this.statusBarItemText = `Azure CLI: Waiting for response`;
+                this.statusBarUpdateInterval = setInterval(() => {
+                    if (this.runningCommandCount === 1)
+                    {
+                        this.commandRunningStatusBarItem.text = `${this.statusBarItemText} ${this.statusBarSpinner()}`;
+                    }
+                    else
+                    {
+                        this.commandRunningStatusBarItem.text = `${this.statusBarItemText} [${this.runningCommandCount}] ${this.statusBarSpinner()}`;
+                    }
+                }, 50);
+            }
+            this.commandRunningStatusBarItem.show();
+            clearTimeout(this.hideStatusBarItemTimeout);
+
+            this.parsedResult = undefined;
+            this.query = undefined; // TODO
+            return this.findResultDocument()
+                .then(document => window.showTextDocument(document, ViewColumn.Two, true))
+                .then(target => replaceContent(target, JSON.stringify({ 'Running command': command }) + '\n')
+                    .then(() => exec(command))
+                    .then(({ stdout }) => stdout, ({ stdout, stderr }) => JSON.stringify({ stderr, stdout }, null, '    '))
+                    .then(content => replaceContent(target, content)
+                            .then(() => this.parsedResult = JSON.parse(content))
+                            .then(undefined, err => {})
+                    )
+                    .then(() => this.commandFinished(t0))
+                )
+                .then(undefined, console.error);
+        }
+    }
+
+    private GetSelectedCommand(source: TextEditor)
+    {
+        if (source.selection.isEmpty)
+        {
+            var lineNumber = source.selection.active.line;
+            if (source.document.lineAt(lineNumber).text.length === 0)
+            {
+                window.showInformationMessage<any>("Please put the cursor on a line that contains a command.");
+                return "";
+            }
+            
+            // find the start of the command (if necessary)
+            while(!source.document.lineAt(lineNumber).text.trim().toLowerCase().startsWith("az"))
+            {
+                lineNumber--;
+            }
+
+            var command = this.StripComments(source.document.lineAt(lineNumber).text);
+
+            // using backtick (`) as continuation character
+            while (command.trim().endsWith("`"))
+            {
+                // concatenate all lines into a single command
+                lineNumber ++;
+                command = command.replace("`", "") + this.StripComments(source.document.lineAt(lineNumber).text);
```

Comment:
.replace() will replace the first match, but you want to replace the last one.

Reply:
Had to go back and refresh my memory on this one. The intended use is a single instance of the continuation character at the end of the line so there should only be one match. I can write some protection code to handle other cases; do think that's useful?

Reply:
Nevermind; I took another look & did some refactoring